### ISSUE TITLE
Making this library less opinionated

### ISSUE THREAD

**Issue by danbovey** [bvaughn/redux-search#73]
Hi

It's my first time using redux-search today but it took me at least an hour just to get the install steps in the README done. So I hope this feedback is useful:

- I feel like the docs could be made to be more available to a wider audience by creating an example with a vanilla redux setup - and by that I mean without introducing another library (reselect) that maybe I haven't heard about and don't want to install just to fit around your examples.

I'm not sure why this library makes the assumptions that it does when some things are fully-configurable and others just plain aren't. It seems counter-intuitive and makes the library opinionated.

- The `resource.id` field should be configurable so that I can tell redux-search what I want my unique key to be, and not just a hard-coded assumption in the README.

https://github.com/bvaughn/redux-search/blob/fa0845eb47fd05e7a6e40db8ea40f7cfed984691/source/SearchApi.js#L69

- I got an error within the search worker because the resource.id is expected to be a string and it immediately tries to call `.trim` on the resource key without casting it. And without a way to customize the ID field there's no nice way to transform my ID into a string just for this library.

https://github.com/bvaughn/js-worker-search/blob/d61f51552f1aac516c84cbd7ab1771af030cdca7/src/util/SearchUtility.js#L210

I appreciate you probably initially made this library for your own needs, and now that I've got it set up correctly this lib is going to cover my needs perfectly as well, but just these few hiccups occurred when getting set up which I thought would be worth sharing.

**Comment by bvaughn** [comment]
Thanks for the feedback. 😄

Would you be interested in contributing to this project? Either to the docs- or the source itself?

I haven't worked with a Redux project (or this library) in almost two years, and I don't really have the time to ramp back up on it at the moment. I would love to see it improved in the ways you suggest though!

**Comment by danbovey** [comment]
Yeah these are small changes so I'll try and work on a PR when I have some free time myself 😃

**Comment by bvaughn** [comment]
Thank you 😄
